File: tests/ManglingTests.cs
using System;
using NUnit.Framework;

namespace Tests {

	[TestFixture]
	public class ManglingTests {

		[Test]
		public void TestCompression ()
		{
			Compression.Test1 (null, "foo", null, "bar");
		}

		[Test]
		public void TestNamespaced ()
		{
			Ns1.Namespaced.Test1 ();
			Ns1.Namespaced.Test2 (null);
		}

		[Test]
		public void TestNamespaced2 ()
		{
			var cls = new Ns1.Ns2.Namespaced2 ();
			cls.Test1 ();
			cls.Test2 (null);
		}
	}
}

File: tests/Native/ManglingTests.cpp

#include "ManglingTests.h"
#include <stdio.h>

void Compression::Test1 (const Compression* a1, const char* a2, const Compression* a3, const char* a4)
{
	printf ("Compression::Test1");
}

void Ns1::Namespaced::Test1 ()
{
	printf ("Ns1::Namespaced::Test1");
}

void Ns1::Namespaced::Test2 (const Compression* a1)
{
	printf ("Ns1::Namespaced::Test2");
}

Ns1::Ns2::Namespaced2::Namespaced2 ()
{
	printf ("Ns1::Ns2::Namespaced2::Namespaced2");
}

void Ns1::Ns2::Namespaced2::Test1 ()
{
	printf ("Ns1::Ns2::Namespaced2::Test1");
}

Ns1::Ns2::Namespaced2* Ns1::Ns2::Namespaced2::Test2 (Compression* a1)
{
	printf ("Ns1::Ns2::Namespaced2::Test2");
}